demos/common/python/models/model.py

```diff
@@ -37,3 +42,9 @@ def set_batch_size(self, batch):
             new_shape = [batch] + self.net.input_info[input_layer].input_data.shape[1:]
             shapes.update({input_layer: new_shape})
         self.net.reshape(shapes)
+
+    def normalize(self, inputs):
+        normalized_inputs = np.zeros(inputs.shape, dtype=np.float32)
+        for i in range(inputs.shape[1]):
+            normalized_inputs[:, i, :, :] = (inputs[:, i, :, :] - self.mean_values[i]) / self.scale_values[i]
+        return normalized_inputs
```

Comment:
```suggestion
        return (inputs - self.mean_values[:, None, None]) / self.scale_values[:, None, None]
```
Just force dtype of `self.mean_values` and `self.scale_values` to `np.float32`.
Since this is one line of code, I don't think you need a wrapping function. Or you can put there checks determining if rescaling should be performed.
If you keep the function, may be name it `scaleshift()` since the image itself isn't always normed, norm may depend on the whole dataset.

Reply:
Updated.

Reply:
Checking only `scale_values` is None in `scaleshift()` is strange. `mean_values` and `scale_values` are very similar so they should go together, but here you split them by checking `mean_values` in `preprocess()` and `scale_values` in `scaleshift()`. Move `self.mean_values is not None` into `scaleshift()`.

Reply:
I simplified the code, so all cases are considered without additional conditions. Now it seems to be consistent with documentation and MO.

Reply:
There is no reason to support `None` now. It should be (1.0, 1.0, 1.0) and (0.0, 0.0, 0.0) as default key values. This is more explicit.

MO probably has `None` which doesn't add rescaling at all. So actually the solution with unconditional rescaling is inconsistent with MO. But I think it doesn't matter much for us.

Reply:
Changed default values.